flutter/Dockerfile

```diff
@@ -1,25 +1,26 @@
-FROM gitpod/workspace-full-vnc
+FROM gitpod/workspace-full
 
+USER root
 
-ENV ANDROID_HOME=/home/gitpod/android-sdk-linux \
-    FLUTTER_HOME=/home/gitpod/flutter \
-    PATH=/usr/lib/dart/bin:$FLUTTER_HOME/bin:$ANDROID_HOME/tools:$PATH
+SHELL [ "/bin/bash", "-c" ]
 
-USER root
+### Install Flutter
+# dependencies
+RUN set -ex; \
+    apt-get update; \
+    apt-get install -y libglu1-mesa
```

Comment:
Our usual snippet for installing things with `apt-get` is:

```Dockerfile
USER gitpod
RUN sudo apt-get update && \
    sudo apt-get install -y libglu1-mesa && \
    sudo rm -rf /var/lib/apt/lists/*
```

I.e., user `gitpod` with `sudo` as mentioned above, and removing left-over data from `apt-get` to keep the image size small. This has historically worked well.

I'm not against changing this everywhere, but I'd like to understand your approach first. Specifically, what is the benefit of using `set -ex` and separating commands with `;` here?

Reply:
As in https://github.com/gitpod-io/workspace-images/pull/165#issuecomment-587039533, `set -e` with `;` behaves the same as `&&`. 

The main motivation comes from the following:

In my opinion we should always use `set -x` in any `RUN` statement with multi commands. This way, in a case where the build fails, we would know exactly what command failed in a set of them just looking in the logs. Much easier to debug. Take a look please:
[![asciicast](https://asciinema.org/a/lh4qb5lEWQvWQh0aaAjbzj0Ys.svg)](https://asciinema.org/a/lh4qb5lEWQvWQh0aaAjbzj0Ys)

The second motivation is that `;` is the designed character to run multiple commands in a single line (which is how the multi-line in `RUN` is interpreted by the shell). ([reference](https://www.tldp.org/LDP/abs/html/special-chars.html)).

The third motivation is since we would use `set -x` already, adding a extra `e` character to it is easier than adding `&&` to every new command, considering that `;` is shorter than `&&`.

By the way [Dockerfile reference](https://docs.docker.com/engine/reference/builder/#run) uses `;`, despite they don't do any kind of error catch (`set -e` I mean).

Reply:
PS: I don't mind in settle in your current way. I just wanted to open the discussion for something I think is better.

Reply:
Thanks a lot for explaining your preference of `set -e` / `;` over `&&`!

It makes sense to me, and I'm happy to start using it in more places in this repo. 👍